core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileObject.java

```diff
@@ -126,132 +168,163 @@ public ConnectionFileObject( AbstractFileName name, ConnectionFileSystem fs,
+    if ( resolvedFileObject != null ) {
+      resolvedFileObject.holdObject( strongRef );
+    } else {
+      super.holdObject( strongRef );
+    }
   }
 
-  @Override public boolean isAttached() {
-    return resolvedFileObject.isAttached();
+  @Override
+  public boolean isAttached() {
+    return resolvedFileObject == null || resolvedFileObject.isAttached();
   }
 
-  @Override public boolean isContentOpen() {
-    return resolvedFileObject.isContentOpen();
+  @Override
+  public boolean isContentOpen() {
+    return resolvedFileObject != null && resolvedFileObject.isContentOpen();
   }
 
-  @Override public boolean isExecutable() throws FileSystemException {
-    return resolvedFileObject.isExecutable();
+  @Override
+  public boolean isExecutable() throws FileSystemException {
+    return resolvedFileObject != null && resolvedFileObject.isExecutable();
   }
 
-  @Override public boolean isFile() throws FileSystemException {
-    return resolvedFileObject.isFile();
+  @Override
+  public boolean isFile() throws FileSystemException {
+    return resolvedFileObject != null && resolvedFileObject.isFile();
   }
 
-  @Override public boolean isFolder() throws FileSystemException {
-    return resolvedFileObject.isFolder();
+  @Override
+  public boolean isFolder() throws FileSystemException {
+    return resolvedFileObject != null ? resolvedFileObject.isFolder() : isPvfsRoot();
   }
 
-  @Override public boolean isHidden() throws FileSystemException {
-    return resolvedFileObject.isHidden();
+  @Override
+  public boolean isHidden() throws FileSystemException {
+    return resolvedFileObject != null && resolvedFileObject.isHidden();
   }
 
-  @Override public boolean isReadable() throws FileSystemException {
-    return resolvedFileObject.isReadable();
+  @Override
+  public boolean isReadable() throws FileSystemException {
+    return resolvedFileObject != null && resolvedFileObject.isReadable();
   }
 
-  @Override public boolean isWriteable() throws FileSystemException {
-    return resolvedFileObject.isWriteable();
+  @Override
+  public boolean isWriteable() throws FileSystemException {
+    // TODO: should let the base class handle it and then implement doWriteable instead.
+    return ensureResolvedFileObject().isWriteable();
   }
 
-  @Override public void moveTo( FileObject destFile ) throws FileSystemException {
-    resolvedFileObject.moveTo( destFile );
+  @Override
+  public void moveTo( FileObject destFile ) throws FileSystemException {
+    ensureResolvedFileObject().moveTo( destFile );
   }
 
-  @Override public void refresh() throws FileSystemException {
+  @Override
+  public void refresh() throws FileSystemException {
     if ( resolvedFileObject != null ) {
       resolvedFileObject.refresh();
     }
   }
 
+  @Nullable
   public FileObject getResolvedFileObject() {
     return resolvedFileObject;
   }
 
+  @NonNull
+  protected AbstractFileObject<?> ensureResolvedFileObject() throws FileSystemException {
```

Comment:
Trying to provide the most specific possible error message to satisfy https://hv-eng.atlassian.net/browse/BACKLOG-40732.

May still be improved to have a separate error for an undefined connection.